controllers/authController.js

```diff
@@ -37,36 +45,128 @@ exports.login = async (req, res, next) => {
 
 exports.register = async (req, res, next) => {
   try {
-    const { userName, phoneNumber, password, confirmPassword } = req.body;
+    const result = await sequelize.transaction(async (t) => {
+      const {
+        username,
+        password,
+        confirmPassword,
+        role,
+        email,
+        phoneNumber,
+        educationArray,
+        experienceArray,
+        skillArray,
+        companyName,
+        websiteLink,
+        overView,
+        address,
+        location,
+      } = req.body;
 
-    if (!phoneNumber) {
-      createError("phone number is require", 400);
-    }
-    if (!password) {
-      createError("password is require", 400);
-    }
-    if (password !== confirmPassword) {
-      createError("password did not match", 400);
-    }
+      const stockPic = {};
 
-    const isPhoneNumber = validator.isMobilePhone(phoneNumber + "");
-    if (!isPhoneNumber) {
-      createError("Invalid phone number", 400);
-    }
+      if (req.files?.profilePic) {
+        const result = await cloudinary.upload(req.files.profilePic[0].path);
+        stockPic.profilePic = result.secure_url;
+      }
+      if (req.files?.coverPic) {
+        const result = await cloudinary.upload(req.files.coverPic[0].path);
+        stockPic.coverPic = result.secure_url;
+      }
 
-    const hashedPassword = await bcrypt.hash(password, 12);
-    const user = await User.create({
-      userName,
-      phoneNumber,
-      password: hashedPassword,
-    });
+      if (!phoneNumber) {
+        createError("phone number is require", 400);
+      }
+      if (!password) {
+        createError("password is require", 400);
+      }
+      if (password !== confirmPassword) {
+        createError("password did not match", 400);
+      }
 
-    const token = createToken({ id: user.id });
+      const isPhoneNumber = validator.isMobilePhone(phoneNumber + "");
```

Comment:
ขอ comment ส่วน validate phone number ว่า มนัตรวจสอบยังไง

Reply:
Nong ping the logic at validatorismoblePhone  it will check country and startnumber from 66||0 and have 9 carector